src/ConnectionAutomaton.cpp

```diff
@@ -568,8 +561,75 @@ void ConnectionAutomaton::onConnectionFrame(
   }
 }
 
+void ConnectionAutomaton::handleStreamFrame(
+    StreamId streamId,
+    FrameType frameType,
+    std::unique_ptr<folly::IOBuf> serializedFrame) {
+  auto it = streamState_->streams_.find(streamId);
+  if (it == streamState_->streams_.end()) {
+    handleUnknownStream(streamId, frameType, std::move(serializedFrame));
+    return;
+  }
+  auto& automaton = it->second;
+
+  switch (frameType) {
+    case FrameType::REQUEST_N: {
+      Frame_REQUEST_N frame_request_n;
+      if (!deserializeFrameOrError(frame_request_n,
+                                   std::move(serializedFrame))) {
+        return;
+      }
+      automaton->handleRequestN(frame_request_n.requestN_);
+      break;
+    }
+    case FrameType::CANCEL: {
+      automaton->handleCancel();
+      break;
+    }
+    case FrameType::PAYLOAD: {
+      Frame_PAYLOAD frame_payload;
+      if (!deserializeFrameOrError(frame_payload,
+                                   std::move(serializedFrame))) {
+        return;
+      }
+      automaton->handlePayload(std::move(frame_payload.payload_),
+                               frame_payload.header_.flagsComplete(),
+                               frame_payload.header_.flagsNext());
+      break;
+    }
+    case FrameType::ERROR: {
+      Frame_ERROR frame_error;
+      if (!deserializeFrameOrError(frame_error,
+                                   std::move(serializedFrame))) {
+        return;
+      }
+      automaton->handleError(frame_error.payload_.moveDataToString());
```

Comment:
You'll need to give it an actual exception, `std::runtime_error("the_string")` is probably fine for now.